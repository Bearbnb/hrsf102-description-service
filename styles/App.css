.container {
  max-width: 600px;
  margin: auto;
  font-family: Circular,"Helvetica Neue",Helvetica,Arial,sans-serif;
}