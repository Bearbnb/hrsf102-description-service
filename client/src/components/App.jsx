import React from 'react';
import axios from 'axios';

import { getRandomInt } from '../../../seedDatabase/helpers';

import HomeHighlights from './HomeHighlights.jsx';
import HeaderContainer from './HeaderContainer.jsx';
import AmenitiesDisplay from './AmenitiesDisplay.jsx';
import Description from './Description.jsx';
import styles from '../../../styles/App.css';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      details: {
        id: 0,
        name: '',
        ownerName: '',
        ownerPicture: '',
        location: '',
        livingSpace: '',
        guests: 0,
        bedrooms: 0,
        beds: 0,
        baths: 0,
        description: '',
        homehighlights: [],
        displayAmenities: [],
        amenities: {},
      },
      showModal: false,
    };
    this.showAmenities = this.showAmenities.bind(this);
    this.hideAmenities = this.hideAmenities.bind(this);
  }

  componentWillMount() {
    const listing = getRandomInt(1, 101);
    axios.get('/listings', {
      params: {
        id: listing,
      },
    }).then((res) => {
      this.setState({
        details: res.data,
      });
    }).catch((err) => {
      console.log(err);
    });
  }

  showAmenities() {
    this.setState({
      showModal: true,
    });
  }

  hideAmenities() {
    console.log(this);
    this.setState({
      showModal: false,
    });
  }

  render() {
    const { details } = this.state;
    const {
      name,
      ownerName,
      ownerPicture,
      location,
      livingSpace,
      guests,
      bedrooms,
      beds,
      baths,
      description,
      homehighlights,
      displayAmenities,
      amenities,
    } = details;
    const { showModal } = this.state;

    if (ownerName === '') {
      return <div />;
    }

    return (
      <div className={styles.container}>
        <HeaderContainer
          guests={guests}
          bedrooms={bedrooms}
          beds={beds}
          baths={baths}
          name={name}
          ownerName={ownerName}
          ownerPicture={ownerPicture}
          location={location}
          livingSpace={livingSpace}
        />
        <HomeHighlights
          homehighlights={homehighlights}
        />
        <Description
          description={description}
        />
        <AmenitiesDisplay
          displayAmenities={displayAmenities}
          amenities={amenities}
          showModal={showModal}
          showAmenities={this.showAmenities}
          hideAmenities={this.hideAmenities}
        />
      </div>
    );
  }
}

export default App;
