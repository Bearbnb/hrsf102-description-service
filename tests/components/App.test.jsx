import { shallow, mount } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import App from '../../client/src/components/App';

describe('<App />', () => {
  it('should render properly', () => {
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call componentDidMount once', () => {
    const spy = jest.spyOn(App.prototype, 'componentWillMount');
    const wrapper = mount(<App />);

    wrapper.instance().componentWillMount();
    expect(spy).toHaveBeenCalled();
  });
});
