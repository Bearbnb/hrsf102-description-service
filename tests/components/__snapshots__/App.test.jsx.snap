// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render properly 1`] = `
<div
  className="container"
>
  <HeaderContainer
    baths={0}
    bedrooms={0}
    beds={0}
    guests={0}
    livingSpace=""
    location=""
    name=""
    ownerName=""
  />
  <HomeHighlights
    homehighlights={Array []}
  />
  <Description
    description=""
  />
  <AmenitiesDisplay
    amenities={Object {}}
    displayAmenities={Array []}
    hideAmenities={[Function]}
    showAmenities={[Function]}
    showModal={false}
  />
</div>
`;
